charts/tests/ut/cp_statefulset.bats

```diff
@@ -0,0 +1,547 @@
+#!/usr/bin/env bats
+
+load "utils.bash"
+
+export TEMPLATE_UNDER_TEST="templates/statefulset.yaml"
+
+setup_file () {
+    cd "$(cp_chart_dir)" || exit
+    helm dependency update .
+}
+
+@test "Control Plane StatefulSet: Name consists of the release name, template name and index" {
+    template
+    assert_query_equal '.metadata.name' "release-name-xrd-control-plane"
+}
+
+@test "Control Plane StatefulSet: Name can be overridden with fullnameOverride" {
+    template --set 'fullnameOverride=xrd-test'
+    assert_query_equal '.metadata.name' "xrd-test"
+}
+
+@test "Control Plane StatefulSet: Name can be overridden with nameOverride" {
+    template --set 'nameOverride=xrd-test'
+    assert_query_equal '.metadata.name' "release-name-xrd-test"
+}
+
+@test "Control Plane StatefulSet: Namespace is default" {
+    template
+    assert_query_equal '.metadata.namespace' "default"
+}
+
+@test "Control Plane StatefulSet: No annotations are set by default" {
+    template
+    assert_query '.metadata.annotations | not'
+}
+
+@test "Control Plane StatefulSet: Global annotations and commonAnnotations can be set" {
+    template --set 'global.annotations.foo=bar' --set 'commonAnnotations.baz=baa'
+    assert_query_equal '.metadata.annotations.foo' "bar"
+    assert_query_equal '.metadata.annotations.baz' "baa"
+}
+
+@test "Control Plane StatefulSet: Annotations, global annotations and commonAnnotations can be set" {
+    template \
+        --set 'global.annotations.foo=bar' \
+        --set 'commonAnnotations.baz=baa' \
+        --set 'annotations.qux=quux'
+    assert_query_equal '.metadata.annotations.foo' "bar"
+    assert_query_equal '.metadata.annotations.baz' "baa"
+    assert_query_equal '.metadata.annotations.qux' "quux"
+}
+
+@test "Control Plane StatefulSet: Recommended labels are set" {
+    template
+    assert_query_equal '.metadata.labels."app.kubernetes.io/name"' "xrd-control-plane"
+    assert_query_equal '.metadata.labels."app.kubernetes.io/instance"' "release-name"
+    assert_query_equal '.metadata.labels."app.kubernetes.io/managed-by"' "Helm"
+    assert_query '.metadata.labels | has("app.kubernetes.io/version")'
+    assert_query '.metadata.labels | has("helm.sh/chart")'
+}
+
+@test "Control Plane StatefulSet: Global labels and commonLabels can be set" {
+    template --set 'global.labels.foo=bar' --set 'commonLabels.baz=baa'
+    assert_query_equal '.metadata.labels.foo' "bar"
+    assert_query_equal '.metadata.labels.baz' "baa"
+}
+
+@test "Control Plane StatefulSet: Labels can be added" {
+    template --set 'labels.foo=bar'
+    assert_query_equal '.metadata.labels.foo' "bar"
+}
+
+@test "Control Plane StatefulSet: Replicas is one" {
+    template
+    assert_query_equal '.spec.replicas' "1"
+}
+
+@test "Control Plane StatefulSet: serviceName is the name of the StatefulSet" {
```

Comment:
This is weird, I didn't know this existed :).

Pretty sure we don't create a Service, so I have no idea how/why this works.  The docs suggest this Service must exist.  I can check with Matt on this.

Reply:
(no need to block this PR)